src/PFRICH_geo_v1.cpp

```diff
@@ -0,0 +1,1407 @@
+   description.add(Constant("PFRICH_filter_material", filterMat.ptr()->GetName(), "string"));
+   description.add(Constant("PFRICH_gasvol_material", gasvolMat.ptr()->GetName(), "string"));
+
+   float _FLANGE_EPIPE_DIAMETER_ = 10.53;   // in cm
+   float _FLANGE_HPIPE_DIAMETER_ = 4.47;    // in cm
+   float _FLANGE_HPIPE_OFFSET_ = 6.76;    // in cm
+   float clearance = 0.5;                   // in cm
+   float length = 45;                       // in cm
+
+
+/////*--------------------------------------------------*/
+//  /// Inner mirror cone
+//  //
+//  // FIXME: do I really care about re-using the same names for these shapes?;
+//  // A wedge bridging two cylinders;
+
+  Tube   eflange(0.0, _FLANGE_EPIPE_DIAMETER_/2 + clearance, length/2);
+  Tube   hflange(0.0, _FLANGE_HPIPE_DIAMETER_/2 + clearance, length/2);
+
+  double r0 = _FLANGE_EPIPE_DIAMETER_/2 + clearance;
+  double r1 = _FLANGE_HPIPE_DIAMETER_/2 + clearance;
+  double L = _FLANGE_HPIPE_OFFSET_;
+  double a = r0*L/(r0-r1);
+  double b = r0*r0/a;
+  double c = r1*(a-b)/r0;
+
+  // GEANT variables to define G4Trap;
+  double pDz = length/2, pTheta = 0.0, pPhi = 0.0, pDy1 = (a - b - c)/2, pDy2 = pDy1;
+  double pDx1 = sqrt(r0*r0 - b*b), pDx2 = pDx1*r1/r0, pDx3 = pDx1, pDx4 = pDx2, pAlp1 = 0.0, pAlp2 = 0.0;
+
+  Trap wedge(pDz, pTheta, pPhi, pDy1, pDx1, pDx2, pAlp1, pDy2, pDx3, pDx4, pAlp2);
+
+  // flange_shape = new G4UnionSolid("Tmp", eflange, hflange, 0, G4ThreeVector(-_FLANGE_HPIPE_OFFSET_, 0.0, 0.0));
+//  Rotation3D  rot(RotationZYX(0, M_PI, 0));
+//  Transform3D transform(rot, Position(0, 0, -149));
+
+  UnionSolid flange_shape(eflange, hflange, Position(-_FLANGE_HPIPE_OFFSET_, 0.0, 0.0));
+
+  Rotation3D rZ(RotationZYX(M_PI/2.0, 0.0, 0.0));
+
+  Transform3D transform_flange(rZ, Position(-b - pDy1, 0.0, 0.0));
+
+  UnionSolid flange_final_shape(flange_shape, wedge, transform_flange);
+
+//  Volume flangeVol(detName + "_flange", flange_shape, mirrorMat);
+  Volume flangeVol(detName + "_flange", flange_final_shape, mirrorMat);
+  flangeVol.setVisAttributes(mirrorVis);
+
+   PlacedVolume flangePV = volume.placeVolume(flangeVol, Position(0, 0, 0));
+   DetElement   flangeDE(sdet, "flange_de", 0);
+   flangeDE.setPlacement(flangePV);
+
+  ///*--------------------------------------------------*/
+  /// SENSOR MODULE LOOP ------------------------
+  /* cartesian tiling loop
+   * - start at (x=0,y=0), to center the grid
+   * - loop over positive-x positions; for each, place the corresponding negative-x sensor too
+   * - nested similar loop over y positions
+   */
+
+   Box    sensorSolid(sensorSide / 2., sensorSide / 2., sensorThickness / 2.);
+   Volume sensorVol(detName + "_sensor", sensorSolid, sensorMat);
+   sensorVol.setVisAttributes(sensorVis);
+
+   // -- Mirrors ---------------------------------------------------------------------------------
+   //
+   // Some "standard" value applied to all mirrors;
+   double _MIRROR_REFLECTIVITY_ = 0.90;
```

Comment:
Local variables should (by convention) not use all capital letters; that's reserved for global variables. Variables should not start with underscores. That's uncommon in C++ (as opposed to python) and may lead to conflicts since cstdlib functions start with an underscore to avoid conflicts with user-defined variables that therefore should not (e.g. https://en.cppreference.com/w/cpp/utility/program/_Exit).

Reply:
Understood. However, Alexander has this type of definition currently inside of his standalone model, I suggest keeping the naming convention as is for now (during the conversion stage), I will change it as soon as the conversion finishes, so there is no confusion between the two sets of geometries.

Reply:
> as soon as the conversion finishes

As long as there is no timeline for retiring the standalone code (and therefore a timeline for "the conversion finishes"), we should work under the assumption that the standalone code will be with us for a long time. Therefore, I encourage you to consider making this change to the naming scheme you are using for variables (or setting a timeline by when it will be done, creating an issue, and adding it to a milestone).

Reply:
Acknowledged. I will push to change the variable names in the standalone code, so things can be consistent between the two sets of code, I will for sure bring this up for the discussion.